airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -556,7 +556,9 @@ def check_for_partition(self, schema: str, table: str, partition: str) -> bool:
         True
         """
         with self.metastore as client:
-            partitions = client.get_partitions_by_filter(schema, table, partition, 1)
+            partitions = client.get_partitions_by_filter(
+                schema, table, partition, HiveMetastoreHook.MAX_PART_COUNT
```

Comment:
Is this a breaking change?
is 1 is not desirable in all cases? (if there are valid use cases for 1 then maybe we should make the 4th parameter (limit) configurable thus allowing each user to set whatever value he wishes?

Reply:
good call. it isn't a breaking change since it is invoked inside `check_for_partition`, which returns `bool`.

Reply:
Correct! Nice one :).